lib/matplotlib/axis.py

```diff
@@ -1810,6 +1816,39 @@ def convert_units(self, x):
                                          f'units: {x!r}') from e
         return ret
 
+    def get_converter(self):
+        """
+        Get the unit converter for axis.
+
+        Returns
+        -------
+        `~matplotlib.units.ConversionInterface` or None
+        """
+        return self.converter
+
+    def set_converter(self, converter):
+        """
+        Set the unit converter for axis.
+
+        Parameters
+        ----------
+        converter : `~matplotlib.dates.ConversionInterface`
+        """
+        self._set_converter(converter)
+        self._converter_is_explicit = True
+
+    def _set_converter(self, converter):
+        if self.converter == converter:
+            return
+        if self._converter_is_explicit:
+            raise RuntimeError("Axis already has an explicit converter set")
+        elif self.converter is not None:
+            _api.warn_external(
+                "This axis already has an converter set and "
+                "is updating to a potentially incompatible converter"
+            )
+        self.converter = converter
```

Comment:
We should not have a public `converter` attribute alongside `set_converter` / `get_converter`. This bypasses safeguards and is confusing.

Turing `converter` into a property and properly documenting it would be an alternative to `set_converter` / `get_converter`. But given the prevalence of `get_*`/`set_*` in our API, I'm inclinded to accept the methods and rather make the attribute private.

Reply:
I agree we should privitize it.